test/upgrade/probe_test.go

```diff
@@ -33,26 +32,28 @@ import (
 
 var successFraction = flag.Float64("probe.success_fraction", 1.0, "Fraction of probes required to pass during upgrade.")
 
-const pipe = "/tmp/prober-signal"
-
 func TestProbe(t *testing.T) {
+	t.Parallel()
 	// We run the prober as a golang test because it fits in nicely with
 	// the rest of our integration tests, and AssertProberDefault needs
 	// a *testing.T. Unfortunately, "go test" intercepts signals, so we
 	// can't coordinate with the test by just sending e.g. SIGCONT, so we
 	// create a named pipe and wait for the upgrade script to write to it
 	// to signal that we should stop probing.
-	if err := syscall.Mkfifo(pipe, 0666); err != nil {
-		t.Fatal("Failed to create pipe:", err)
-	}
-	defer os.Remove(pipe)
+	createPipe(pipe, t)
 
 	clients := e2e.Setup(t)
 	names := test.ResourceNames{
 		Service: "upgrade-probe",
 		Image:   test.PizzaPlanet1,
 	}
-	defer test.TearDown(clients, &names)
+
+	cleanup := func() {
+		test.TearDown(clients, &names)
+		os.Remove(pipe)
```

Comment:
I think we can now move this to the `createPipe` function and use `test.EnsureCleanup` in there.